pkg/registry/resolver.go

```diff
@@ -0,0 +1,54 @@
+package api
+
+import (
+	api "github.com/openshift/ci-tools/pkg/api"
+	types "github.com/openshift/ci-tools/pkg/steps/types"
+)
+
+// Registry will hold all the registry information needed to convert between the
+// user provided configs referencing the registry and the internal, complete
+// represenation
+type Registry struct{}
+
+func (r Registry) Resolve(config api.MultiStageTestConfiguration) (types.TestFlow, []error) {
+	var resolveErrors []error
+	expandedFlow := types.TestFlow{}
+	for _, external := range config.Pre {
+		newStep, errs := toInternal(external)
+		if errs != nil {
+			resolveErrors = append(resolveErrors, errs...)
+			continue
+		}
+		expandedFlow.Pre = append(expandedFlow.Pre, newStep)
+	}
+	for _, external := range config.Test {
+		newStep, errs := toInternal(external)
+		if errs != nil {
+			resolveErrors = append(resolveErrors, errs...)
+			continue
+		}
+		expandedFlow.Test = append(expandedFlow.Test, newStep)
+	}
+	for _, external := range config.Post {
+		newStep, errs := toInternal(external)
+		if errs != nil {
+			resolveErrors = append(resolveErrors, errs...)
+			continue
+		}
+		expandedFlow.Test = append(expandedFlow.Test, newStep)
+	}
+	if resolveErrors != nil {
+		return types.TestFlow{}, resolveErrors
+	}
+	return expandedFlow, nil
+}
+
+func toInternal(input api.TestStep) (types.TestStep, []error) {
+	newStep := types.TestStep{}
```

Comment:
nit: this could be

```go
return types.TestStep{
    Name: input.Name,
    Image: input.Image,
    Commands: input.Commands,
    ArtifactDir: input.ArtifactDir,
    Resources: input.Resources,
}
```